### PR TITLE
Updated addon.ini

### PR DESCRIPTION
**Pull request by FransBouma**
updated igcs connector addon reference to latest version

### CODE REVIEW

File: Addons.ini [hunk 1]
```diff
@@ -76,7 +76,7 @@ RepositoryUrl=https://github.com/4lex4nder/ReshadeEffectShaderToggler
 [11]
 PackageName=IGCS Connector by Otis_Inf
 PackageDescription=Connects ReShade with a photomode mod built with IGCS to exchange data and issue actions
-DownloadUrl64=https://github.com/FransBouma/IgcsConnector/releases/download/v232/IGCSConnector_v232.zip
+DownloadUrl64=https://github.com/FransBouma/IgcsConnector/releases/download/v232/IGCSConnector_v240.zip
```
Comment (cot6) [inline Addons.ini]: `https://github.com/FransBouma/IgcsConnector/releases/download/v240/IGCSConnector_v240.zip`

Comment (FransBouma) [reply Addons.ini]: d'oh! will fix!

Comment (FransBouma) [reply Addons.ini]: fixed

**Comment by FransBouma** [comment]
Updated the PR with the release of IGCS Connector v2.5.0.

**Comment by FransBouma** [comment]
Updated the PR with the release of IGCS Connector v2.5.1

It would have been really great if the author of the installer had reached out to me so I had known up front an ini file with a dependency on a hardcoded URL on my repo would be introduced. This isn't a system that really works. I'm accused of breaking things because I release my stuff on a single release. That wouldn't be so bad but this ini file isn't merged very often. And that might have good reasons, we're all human and it's not a 24/7 job. However the installer depends on the URLs being right and if they're not, because this ini is outdated, it's perhaps not the best system for this purpose. Older releases might work around a 404 but it also introduces the fact that users then will use older/outdated versions while newer ones were available. 

I wondered why wasn't I contacted? Then perhaps a better system would have been created (e.g. the installer pulls a file from my repo with the url so it's on me if it breaks, a requirement IF I want to have my addon in the list that I have to have a stable release URL? ). We're all in the same discord, a simple tag would have been enough, a thread with some discussions and we would have come up with something that might be better for all involved than this ini file. 

Now I was confronted with errors being reported I didn't know where they were coming from (users told me they got a 404 but not where, while it turned out to be the reshade installer) nor how to fix them (apparently there was this ini file) and when I fix it, the ini file isn't merged very often (so users get old code / addons or a 404. Again, totally ok, but it does have the result that the installer gives problems). 

Now there's drama over nothing which all could have been avoided. 

Edited for clarity.

**Comment by crosire** [comment]
Add-on downloading were just a simple copy of the existing effect repository downloading with slight modifications. It was a trivial addition, no thought went into it, nor was it inteded to be a big thing that could cause discourse. And I'm not monitoring ReShade stuff 24/7, so sometimes it can take a bit before I notice and have time or muse to do something about it. I'm sorry you got confronted with errors that have no direct relation to you, that was never an intention of course.

**Comment by FransBouma** [comment]
@crosire Yeah no worries. I was thinking perhaps a file in the repo with the URL could be a solution? Tho I don't expect a lot of changes in the near future in my addons, but as the list of addons likely grows, so does the number of changes to this ini file... With a file where the addon to load from, it's upon the maintainer of the addon to update it. 

Downside is of course that it could also be an attack vector and could offer a payload with whatever content it points to... :(

**Comment by crosire** [comment]
Could be done, does require that file to be in a specific format though, while currently add-ons can be delivered from arbitrary sources (the download link doesn't have to point to GitHub) without requiring much action from the add-on developers (unless there is an update and I'm not fast enough to notice and update the list myself, as happened here).
I think the problem is mainly just that the download links for GitHub releases are not static, e.g. hosting the binaries in the Git repository itself or on some other web service with static links, or even just a redirect link that you control which points to the latest GitHub release would also solve it.

**Comment by Mortalitas** [comment]
You can technically get a pseudo-permalink to the latest release of a file by always naming a given release file in a specific way, e.g. the current link would be:
https://github.com/FransBouma/IgcsConnector/releases/latest/download/IGCSConnector_v251.zip

But if the file were named `IGCSConnector.zip` the permalink would be:
https://github.com/FransBouma/IgcsConnector/releases/latest/download/IGCSConnector.zip

**Comment by FransBouma** [comment]
@Mortalitas I want the version in the zip's filename so that already breaks the URL. I didn't know you could use 'latest' as a uniform tag to get the latest release. 

@crosire will think about it, none of them are ideal (for me), but neither is the current setup (for users as they potentially get 404's).

**Pull request merged**